### PR TITLE
chore: libwaku retrieve my enr and adapt golang example

### PR DESCRIPTION
**Pull request by Ivansete-status**
## Description
Add new "get ENR" method available in libwaku

## Changes

- new `waku_get_my_enr` available in `libwaku`
- adapt Golang example to use the new `waku_get_my_enr` function

### CODE REVIEW

File: examples/golang/waku.go [hunk 1]
```diff
@@ -446,6 +451,20 @@ func (self *WakuNode) WakuListenAddresses() (string, error) {
 	return "", errors.New(errMsg)
 }
 
+func (self *WakuNode) WakuGetMyENR() (string, error) {
+	var resp = C.allocResp()
+	defer C.freeResp(resp)
+	C.cGoWakuGetMyENR(self.ctx, resp)
+
+	if C.getRet(resp) == C.RET_OK {
+		var listenAddresses = C.GoStringN(C.getMyCharPtr(resp), C.int(C.getMyCharLen(resp)))
```
Comment (NagyZoltanPeter) [inline examples/golang/waku.go:460]: Maybe a bit misleading to call it listenAddresses and not the ENR.
I remember that is different also in /debug/info API result.

**Review by NagyZoltanPeter — APPROVED** [review]
LGTM, added one little naming observation...

File: examples/golang/waku.go [hunk 2]
```diff
@@ -446,6 +451,20 @@ func (self *WakuNode) WakuListenAddresses() (string, error) {
...
+		return listenAddresses, nil
```
Comment (gabrielmer) [inline examples/golang/waku.go:461]: Shouldn't we change the variable name to `enr` or something like that? looks like we kept the same variable name after copy-pasting from `WakuListenAddresses()`

**Review by gabrielmer — APPROVED** [review]
LGTM, thank you! Added just a tiny question

File: examples/golang/waku.go (hunk 2 above)
Comment (gabrielmer) [reply examples/golang/waku.go:461]: ooh @NagyZoltanPeter added the same comment a few minutes before I did lol. Nice!

Comment (Ivansete-status) [reply examples/golang/waku.go:461]: Thank you so much guys ! I'll change that

**Pull request merged**